models/users.js

```diff
@@ -48,6 +49,14 @@ const addOrUpdate = async (userData, userId = null) => {
         userId: user.docs[0].id,
         incompleteUserDetails: user.docs[0].data().incompleteUserDetails,
       };
+    } else if (!user1.empty) {
+      await userModel.doc(user1.docs[0].id).set(userData, { merge: true });
```

Comment:
what does this merge:true specify?

Reply:
As of now user1 variable checks on github_id, under the else if condition I've mentioned that if that field (github_id) is not empty then do the merging on the basis of that. Answering your question without the **merge: true option**, **.set()** would **replace the entire document** with the userData object, deleting any fields that were not specified in userData and **potentially causing data loss.**